### PR TITLE
[15.0][ADD] website_sale_product_attribute_filter_collapse: New module

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
This module has been created to make independent the functionality of collapse attributes in the attributes section that existed in the website_sale_product_attribute_filter_category module whose functionality was to add the attribute categories in the filter section. Both modules are still supported separately.

Modification of module website_sale_product_attribute_filter_category: https://github.com/OCA/e-commerce/pull/738

The tests depend on:
- [x] #744

cc @Tecnativa TT41385

@chienandalu @CarlosRoca13 please review

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]

**Review by chienandalu — COMMENTED** [review]
This module is mainly a split of functionality of the  former website_sale_attribute_filter_category (now website_sale_product_attribute_filter_category) which included the folding of attributes along with categories. That features is dropped in v15 and thus this module. Document the rationale and respect attribution in the code as much as possible (you can use the blame of the former module)

File: website_sale_product_attribute_filter_collapse/README.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+.. contents::
+   :local:
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+* Go to 'Website > Configuration > Attributes, create one or more with
+  this category and add more than one value for each attribute.
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/README.rst]: What does the category have to do with this feature?

File: website_sale_product_attribute_filter_collapse/README.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+* Go to 'Website > Products > Products' and create a product with this
+  attribute and assign it an attribute value.
+
+Usage
+=====
+
+#. Go to Website Shop.
+#. Active "Attributes & Variants filter" in "Customize" option in top menu.
+#. Active "Filter by collapsible attributes" in "Customize" option in top menu.
+#. Now you can see product attributes filter grouped by categories.
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/README.rst]: By categories? It should be by attributes....

File: website_sale_product_attribute_filter_collapse/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/__init__.py]: Remove this license line as the file is empty

File: website_sale_product_attribute_filter_collapse/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Website Sale Attribute Filter Collapse",
+    "summary": "Allows the attributes of the categories to be folded",
+    "version": "15.0.1.0.0",
+    "category": "Website",
+    "website": "https://github.com/OCA/e-commerce",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/__manifest__.py]: It's `False` by default
```suggestion
```

File: website_sale_product_attribute_filter_collapse/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "installable": True,
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/__manifest__.py]: It's `True` by default
```suggestion
```

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 6]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2023 Tecnativa - Pilar Vargas
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <template
+        id="products_attributes_collapsible"
+        inherit_id="website_sale.products_attributes"
+        name="Filter by collapsible attributes"
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml]:
```suggestion
        name="Collapsible attribute filters"
```

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        active="False"
+        customize_show="True"
+        priority="999"
+    >
+        <!-- Add a new title zone -->
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml]: Title zone?

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+                    aria-hidden="true"
+                />
+                <i
+                    class="fa fa-angle-down"
+                    id="icon_attribute_expanded"
+                    aria-hidden="true"
+                />
+            </a>
+        </xpath>
+        <!-- -->
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml]: :no_mouth:

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        <xpath
+            expr="//ul[hasclass('flex-column')]/t/li/div/strong"
+            position="attributes"
+        >
+            <attribute name="class">o_shop_collapse_attribute_filter</attribute>
+        </xpath>
+        <!-- Move attribute name to the title zone -->
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml]: What title zone?

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        name="Filter by collapsible attributes"
+        active="False"
+        customize_show="True"
+        priority="999"
+    >
+        <!-- Add a new title zone -->
+        <xpath expr="//ul[hasclass('flex-column')]/t/li/div" position="before">
+            <a
+                data-toggle="collapse"
+                class="collapsed"
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml]: Fix default padding
```suggestion
                class="pl-0 collapsed"
```

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+            <attribute name="class" add="d-none" separator=" " />
+        </xpath>
+        <!-- Move 'select' type to a new collapsible zone -->
+        <xpath
+            expr="//ul[hasclass('flex-column')]/t/li/t[contains(@t-if,'select')]/select"
+            position="before"
+        >
+            <div
+                t-attf-id="collapse_v_{{a.id}}"
+                data-type="select"
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml:56]: Fix padding so it looks visually nested
```suggestion
                data-type="select"
                class="pl-4"
```

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+            />
+        </xpath>
+        <!-- Move 'radio' type to a new collapsible zone -->
+        <xpath
+            expr="//ul[hasclass('flex-column')]/t/li/t[contains(@t-if,'radio')]/div"
+            position="before"
+        >
+            <div
+                t-attf-id="collapse_v_{{a.id}}"
+                data-type="radio"
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml:77]:
```suggestion
                data-type="radio"
                class="pl-4"
```

File: website_sale_product_attribute_filter_collapse/views/templates.xml [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+            />
+        </xpath>
+        <!-- Move 'color' type to a new collapsible zone -->
+        <xpath
+            expr="//ul[hasclass('flex-column')]/t/li/t[contains(@t-if,'color')]/t"
+            position="before"
+        >
+            <div
+                t-attf-id="collapse_v_{{a.id}}"
+                data-type="color"
```
Comment (chienandalu) [inline website_sale_product_attribute_filter_collapse/views/templates.xml:98]:
```suggestion
                data-type="color"
                class="pl-4"
```

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: website_sale_product_attribute_filter_collapse/views/templates.xml (hunk 9 above)
Comment (pilarvargas-tecnativa) [reply website_sale_product_attribute_filter_collapse/views/templates.xml]: To the title zone added above, I'll make it clearer in the comment :)

**Comment by pilarvargas-tecnativa** [comment]
@chienandalu I have made all the suggested changes and also explained the purpose of this new module, thank you :)

**Pull request merged**